test/e2e/single_event_test.go

```diff
@@ -28,77 +28,60 @@ import (
-		loggerPodName    = "e2e-singleevent-logger-pod"
-	)
+func singleEvent(t *testing.T, encoding string) {
+	channelName := "e2e-singleevent-" + encoding
+	senderName := "e2e-singleevent-sender-" + encoding
+	subscriptionName := "e2e-singleevent-subscription-" + encoding
+	loggerPodName := "e2e-singleevent-logger-pod-" + encoding
 
-	clients, cleaner := Setup(t, t.Logf)
-	// verify namespace
-	ns, cleanupNS := CreateNamespaceIfNeeded(t, clients, t.Logf)
-	defer cleanupNS()
-
-	// TearDown() needs to be deferred after cleanupNS(). Otherwise the namespace is deleted and all
-	// resources in it. So when TearDown() runs, it spews a lot of not found errors.
-	defer TearDown(clients, cleaner, t.Logf)
+	clients, ns, provisioner, cleaner := Setup(t, true, t.Logf)
+	defer TearDown(clients, ns, cleaner, t.Logf)
 
 	// create logger pod
 	t.Logf("creating logger pod")
 	selector := map[string]string{"e2etest": string(uuid.NewUUID())}
 	loggerPod := test.EventLoggerPod(loggerPodName, ns, selector)
 	loggerSvc := test.Service(loggerPodName, ns, selector)
-	loggerPod, err := CreatePodAndServiceReady(clients, loggerPod, loggerSvc, ns, t.Logf, cleaner)
+	loggerPod, err := CreatePodAndServiceReady(clients, loggerPod, loggerSvc, t.Logf, cleaner)
 	if err != nil {
 		t.Fatalf("Failed to create logger pod and service, and get them ready: %v", err)
 	}
 
-	// create channel
-
+	// create channel and subscription
 	t.Logf("Creating Channel and Subscription")
-	channel := test.Channel(channelName, ns, test.ClusterChannelProvisioner(test.EventingFlags.Provisioner))
-	t.Logf("channel: %#v", channel)
+	channel := test.Channel(channelName, ns, test.ClusterChannelProvisioner(provisioner))
 	sub := test.Subscription(subscriptionName, ns, test.ChannelRef(channelName), test.SubscriberSpecForService(loggerPodName), nil)
-	t.Logf("sub: %#v", sub)
 
-	if err := WithChannelsAndSubscriptionsReady(clients, &[]*v1alpha1.Channel{channel}, &[]*v1alpha1.Subscription{sub}, t.Logf, cleaner); err != nil {
+	if err := WithChannelsAndSubscriptionsReady(clients, ns, &[]*v1alpha1.Channel{channel}, &[]*v1alpha1.Subscription{sub}, t.Logf, cleaner); err != nil {
 		t.Fatalf("The Channel or Subscription were not marked as Ready: %v", err)
 	}
 
-	// send fake CloudEvent to the first channel
+	// send fake CloudEvent to the channel
 	body := fmt.Sprintf("TestSingleEvent %s", uuid.NewUUID())
 	event := &test.CloudEvent{
 		Source:   senderName,
 		Type:     test.CloudEventDefaultType,
 		Data:     fmt.Sprintf(`{"msg":%q}`, body),
 		Encoding: encoding,
 	}
-	if err := SendFakeEventToChannel(clients, event, channel, ns, t.Logf, cleaner); err != nil {
+	if err := SendFakeEventToChannel(clients, event, channel, t.Logf, cleaner); err != nil {
 		t.Fatalf("Failed to send fake CloudEvent to the channel %q", channel.Name)
 	}
 
 	if err := pkgTest.WaitForLogContent(clients.Kube, loggerPodName, loggerPod.Spec.Containers[0].Name, ns, body); err != nil {
-		clients.Kube.PodLogs(senderName, "sendevent", ns)
```

Comment:
When something goes wrong, these are very useful logs to have.

Reply:
These two functions are only fetching the logs and not printing.
I've added them back and printed them as well.